float64 deltaX
float64 deltaY
